consensus/core/src/core.rs

```diff
@@ -424,43 +435,9 @@ impl Core {
         {
             quorum.add(ancestor.author(), &self.context.committee);
         }
-
         assert!(quorum.reached_threshold(&self.context.committee), "Fatal error, quorum not reached for parent round when proposing for round {}. Possible mismatch between DagState and Core.", clock_round);
 
-        // Ensure that timestamps are correct
-        ancestors.iter().for_each(|block|{
-            // We assume that our system's clock can't go backwards when we perform the check here (ex due to ntp corrections)
-            assert!(block.timestamp_ms() <= block_timestamp, "Violation, ancestor block timestamp {} greater than our timestamp {block_timestamp}", block.timestamp_ms());
-        });
-
-        // Compress the references in the block. We don't want to include an ancestors that already referenced by other blocks
-        // we are about to include.
-        let all_ancestors_parents: HashSet<&BlockRef> = ancestors
-            .iter()
-            .flat_map(|block| block.ancestors())
-            .collect();
-
-        // Keep block refs to propose in a map, so even if somehow a byzantine node managed to provide blocks that don't
-        // form a valid chain we can still pick one block per author.
-        let mut to_propose = BTreeMap::new();
-        for block in &ancestors {
-            if !all_ancestors_parents.contains(&block.reference()) {
-                to_propose.insert(block.author(), block.reference());
-            }
-        }
-
-        assert!(!to_propose.is_empty());
-
-        // always include our last proposed block in front of the vector and make sure that we do not
-        // double insert.
-        let mut result = vec![self.last_proposed_block.reference()];
-        for (authority_index, block_ref) in to_propose {
-            if authority_index != self.context.own_index {
-                result.push(block_ref);
-            }
-        }
```

Comment:
nice clean up on the DAG compression, indeed there isn't much point keeping this